src/python_testing/TC_OCC_3_2.py

```diff
@@ -1,204 +1,240 @@
+
+        attribute_list = await self.read_occ_attribute_expect_success(attribute=attributes.AttributeList)
+        has_pir_timing_attrib = attributes.PIROccupiedToUnoccupiedDelay.attribute_id in attribute_list
+        has_ultrasonic_timing_attrib = attributes.UltrasonicOccupiedToUnoccupiedDelay.attribute_id in attribute_list
+        has_contact_timing_attrib = attributes.PhysicalContactOccupiedToUnoccupiedDelay.attribute_id in attribute_list
+        logging.info(f"Attribute list: {attribute_list}")
+        logging.info(f"--> Has PIROccupiedToUnoccupiedDelay: {has_pir_timing_attrib}")
+        logging.info(f"--> Has UltrasonicOccupiedToUnoccupiedDelay: {has_ultrasonic_timing_attrib}")
+        logging.info(f"--> Has PhysicalContactOccupiedToUnoccupiedDelay: {has_contact_timing_attrib}")
+
+        # min interval = 0, and max interval = 30 seconds
+        attrib_listener = ClusterAttributeChangeAccumulator(Clusters.Objects.OccupancySensing)
+        await attrib_listener.start(dev_ctrl, node_id, endpoint=endpoint_id, min_interval_sec=0, max_interval_sec=30)
+
+        # add Namepiped to assimilate the manual sensor untrigger here
+        self.step("3a")
+        # CI call to trigger off
+        if self.is_ci:
+            self.write_to_app_pipe('{"Name":"SetOccupancy", "EndpointId": 1, "Occupancy": 0}')
+        else:
+            self.wait_for_user_input(prompt_msg="Type any letter and press ENTER after DUT goes back to unoccupied state.")
+
+        self.step("3b")
+        if attributes.Occupancy.attribute_id in attribute_list:
+            initial_dut = await self.read_occ_attribute_expect_success(attribute=attributes.Occupancy)
+            asserts.assert_equal(initial_dut, 0, "Occupancy attribute is still detected state")
+
+        # add Namepiped to assimilate the manual sensor trigger here
+        self.step("3c")
+        # CI call to trigger on
+        if self.is_ci:
+            self.write_to_app_pipe('{"Name":"SetOccupancy", "EndpointId": 1, "Occupancy": 1}')
+        else:
+            self.wait_for_user_input(
+                prompt_msg="Type any letter and press ENTER after the sensor occupancy is triggered and its occupancy state changed.")
+
+        self.step("3d")
+        await_sequence_of_reports(report_queue=attrib_listener.attribute_queue, endpoint_id=endpoint_id, attribute=cluster.Attributes.Occupancy, sequence=[
+                                  1], timeout_sec=post_prompt_settle_delay_seconds)
+
+        self.step("4a")
+        if attributes.HoldTime.attribute_id not in attribute_list:
+            logging.info("No HoldTime attribute supports. Terminate this test case")
+            self.skip_all_remaining_steps("4b")
+
+        self.step("4b")
+        initial_dut = await self.read_occ_attribute_expect_success(attribute=attributes.HoldTime)
+
+        self.step("4c")
+        # write a different a HoldTime attribute value
+        diff_val = 12

```

Comment:
-12 may not be a valid value. Start by writing HoldTimeMin and checking via a *read* that it is OK, then reset the attribute change accumulator before writing HoldTime to HoldTimeMax.
- Then change other attributes to HoldTimeMax and validate they tracked

We are now at the following steps:

- Set HoldTime to HoldTimeMin
- Read HoldTime and check it's HoldTimeMin
- Set to HoldTime HoldTimeMax
  - Check you saw update on the applicable legacy attributes to HoldTimeMax
- For of [PIR, US, PHY] timing attributes, if supported (blocks of steps)
  - Set HoldTime to HoldTimeMin
  - Clear attribute recording history
  - Use [PIR|US|PHY]OccupiedToUnoccupiedDelay to set value HoldTimeMax
  - Check that HoldTime got updated

Reply:
Both reciprocal attribute testing has been added but maybe not exactly what you inferred here.